File: app/src/main/java/com/kaciula/archiman/presentation/screen/home/HomeModule.java
package com.kaciula.archiman.presentation.screen.home;

import com.kaciula.archiman.domain.boundary.SchedulerProvider;
import com.kaciula.archiman.domain.usecases.GetUsers;
import com.kaciula.archiman.injection.ActivityScope;
import dagger.Module;
import dagger.Provides;

@Module
class HomeModule {
  private final HomeContract.View view;

  HomeModule(HomeContract.View view) {
    this.view = view;
  }

  @Provides
  @ActivityScope
  HomeContract.Presenter providePresenter(SchedulerProvider schedulerProvider,
      GetUsers getUsers) {
    return new HomePresenter(view, schedulerProvider, getUsers);
  }
}

File: app/src/main/java/com/kaciula/archiman/presentation/screen/home/HomeModule.kt
package com.kaciula.archiman.presentation.screen.home

import com.kaciula.archiman.domain.usecases.GetUsers
import com.kaciula.archiman.injection.ActivityScope
import com.kaciula.archiman.presentation.util.Elm
import dagger.Module
import dagger.Provides

@Module
class HomeModule(private val view: HomeContract.View) {

    @Provides
    @ActivityScope
    fun providePresenter(getUsers: GetUsers): HomeContract.Presenter
            = HomePresenter(view, Elm(), getUsers)
}
